File: src/main/java/com/machfour/macros/objects/AttrMapping.java
package com.machfour.macros.objects;

import com.machfour.macros.core.*;

public class AttrMapping extends MacrosEntityImpl<AttrMapping> {

    private AttrMapping(ColumnData<AttrMapping> data, ObjectSource objectSource) {
        super(data, objectSource);
    }

    @Override
    public Table<AttrMapping> getTable() {
        return Schema.AttrMappingTable.instance();
    }

    @Override
    public Factory<AttrMapping> getFactory() {
        return factory();
    }
    public static Factory<AttrMapping> factory() {
        return AttrMapping::new;
    }
}

File: src/main/java/com/machfour/macros/objects/AttrMapping.kt
package com.machfour.macros.objects

import com.machfour.macros.core.*
import com.machfour.macros.core.schema.AttrMappingTable
import com.machfour.macros.core.schema.SchemaHelpers

class AttrMapping private constructor(data: ColumnData<AttrMapping>, objectSource: ObjectSource)
    : MacrosEntityImpl<AttrMapping>(data, objectSource) {

    companion object {
        val factory: Factory<AttrMapping> = Factory { dataMap, objectSource -> AttrMapping(dataMap, objectSource) }
        val table: Table<AttrMapping>
            get() = AttrMappingTable.instance
    }

    override val factory: Factory<AttrMapping>
        get() = Companion.factory

    override val table: Table<AttrMapping>
        get() = Companion.table

}
